operator/pkg/controller/workspace/workspace_controller.go

```diff
@@ -225,19 +225,38 @@ func (r *ReconcileWorkspace) Reconcile(request reconcile.Request) (reconcile.Res
 		}
 	}
 	specTFCVariables := MapToTFCVariable(instance.Spec.Variables)
-	updatedVariables, err := r.tfclient.CheckVariables(workspace, specTFCVariables)
+
+	updatedVariables, err := r.tfclient.CheckVariables(workspace, specTFCVariables, instance.Status.SensitiveHashes)
 	if err != nil {
 		r.reqLogger.Error(err, "Could not update variables")
 		return reconcile.Result{}, err
 	}
 
-	if updatedTerraform || updatedVariables || instance.Status.RunID == "" {
-		r.reqLogger.Info("Starting run because template changed", "Organization", organization, "Name", workspace, "Namespace", request.Namespace)
-
-		if err := r.tfclient.UpdateSensitiveBeforeRun(workspace, specTFCVariables); err != nil {
-			r.reqLogger.Error(err, "Could not update Sensitive Varaibles before Run")
+	// Store hashed versions of sensitive values in the status
+	{
+		for _, variable := range updatedVariables {
```

Comment:
This should be moved to after a run to ensure values aren't saved to status and then the run fails for some reason.